### PR TITLE
Safe json stringify

### PR DESCRIPTION
**Pull request by roironn**

### CODE REVIEW

File: packages/adapter-utils/test/utils.test.ts [hunk 1]
```diff
@@ -1250,4 +1251,55 @@ describe('Test utils.ts', () => {
         .toEqual(new InstanceElement('test', configType, { val1: 'test' }))
     })
   })
+
+  describe('JSONSaltoValue', () => {
+    it('should act like JSON.stringiy when there are no circular dependecies', () => {
```
Comment (netama) [inline packages/adapter-utils/test/utils.test.ts:1256]: nit: `stringiy` -> `stringify`

**Comment by ori-moisis** [comment]
The problem is field is not the only thing that can cause circular references in JSON, ReferenceExpressions can also cause it.
So this change won't really solve the whole problem

**Comment by roironn** [comment]
> Instead of this heavy-handed change, how about implementing a [toJSON](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#toJSON_behavior) method on the `Field` class?

Re: We can't use the toJSON method at it looses class information which is needed. for serialization.

File: packages/e2e-credentials-store/src/cli/commands/lease.ts [hunk 2]
```diff
@@ -59,7 +59,7 @@ const commandModule = ({
       writeLine(stderr, 'No lease available')
       return 1
     }
-
+    // eslint-disable-next-line no-restricted-syntax
```
Comment (ori-moisis) [inline packages/e2e-credentials-store/src/cli/commands/lease.ts:62]: maybe this rule should not apply to projects that do not use `Elements`?
(meaning, only projects that depend on `adapter-api` need to have this rule)

Comment (roironn) [reply packages/e2e-credentials-store/src/cli/commands/lease.ts:62]: Good idea. Will add.

File: packages/core/src/local-workspace/static_files_cache.ts [hunk 3]
```diff
@@ -16,6 +16,7 @@
 import path from 'path'
 import { readTextFile, exists, mkdirp, replaceContents, rm, rename } from '@salto-io/file'
 import { staticFiles } from '@salto-io/workspace'
+import { JSONSaltoValue } from '@salto-io/adapter-utils'
```
Comment (royra) [inline packages/core/src/local-workspace/static_files_cache.ts:19]: Last nit: how about naming our function safeJsonStringify?

**Pull request merged**